Develop

javaparamonov77 opened the pull request:
Первоначальная версия проекта была добавлена без PR (в нее входила основная логика работы).
В текущем PR:
1) Добавлены модули вместо папок
2) Скорректирована папочная структура
3) Скорректированы POM-файлы
4) Минорные правки в объектах

paramonov-yo committed:
Корректировка самой первой версии проекта

paramonov-yo committed:
Корректировка самой первой версии проекта

paramonov-yo committed:
Добавлен Spring Boot
Добавлены web-страницы (Client/Web)
Добавлен Spring Security в части сервера
--Требуется продумать работу ATMClientController (в первом приближений он реализован)

RodinDenis commented on client/src/main/java/ru/paramonov/project/client/Application.java:
```diff
@@ -0,0 +1,15 @@
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.security.servlet.SecurityAutoConfiguration;
+
+@EnableAutoConfiguration(exclude = {SecurityAutoConfiguration.class})
```
зачем вы добавили это исключение?

javaparamonov77 replied:
Вообще предполагалось исключение модуля Client из зоны влияния Spring Security (т.е. отсутствие явной авторизации/аутентификации в модуле Client, фактическое использование вышеперечисленного только в модуле Server)

javaparamonov77 replied:
Внес правки по замечаниям (этот же PR)

RodinDenis commented on client/src/main/java/ru/paramonov/project/client/contoller/ATMClientController.java:
```diff
@@ -0,0 +1,50 @@
+    }
+    @GetMapping("/exit")
+    public void exit(){
+        System.exit(0);
```
На уроках мы говорили о последствиях вызова System.exit
Это как "отключение рубильника" и методах контроллера в промышленном коде такое применять нельзя.

javaparamonov77 replied:
Согласен. Исправление будет отражено в новой версии.

javaparamonov77 replied:
Внес правки по замечаниям (этот же PR)

RodinDenis commented on server/pom.xml:
```diff
@@ -0,0 +1,46 @@
+        <dependency>
+            <groupId>org.aspectj</groupId>
+            <artifactId>aspectjweaver</artifactId>
+            <version>1.9.6</version>
```
версии зависимостей лучше выносить в properties

javaparamonov77 replied:
исправил. Будет отражено в новой версии.

javaparamonov77 replied:
Внес правки по замечаниям (этот же PR)

RodinDenis commented on server/src/main/java/ru/paramonov/project/server/controller/ATMAccountServerController.java:
```diff
@@ -0,0 +1,41 @@
+import java.math.BigDecimal;
+import java.util.List;
+
+@Controller
```
если вы используете RestController, то java объекты из ответов будут преобразованы в json самим Spring.

javaparamonov77 replied:
Добавлю исправление в новой версии

javaparamonov77 replied:
Внес правки по замечаниям (этот же PR)

RodinDenis commented:
Нужен контроллер возвращающий json. @RestController конвертирует java объекты в json перед отправкой

paramonov-yo committed:
Исправлены замечания после ревью.
- Добавлен RestController, возвращающий JSON.

paramonov-yo committed:
Внесенные правки:
1) Добавлена отдельная проверка на наличие запрашиваемых данных в БД
2) Скорректирована часть переменных с оберточных на примитивные
3) В классе APIRestController аннотация @Autowired над переменной типа AccountService заменена на инициализацию в конструкторе
4) Скорректированы модификаторы доступа в нескольких классах
5) Исправлена ошибка определения наличия объектов в БД
6) Добавлено определение количества попыток некорректно введенного пользователем ПИН-кода (дается 3 попытки)
